Display the Table of School Data

DelroyGayle opened the pull request:
## 6 - Visualise Data - Delroy Gayle 

## Description

Display the data using sample data from
client/src/pages/schools.json

Used Chakra Package to Create a Table in order to display the school data
The table can be displayed with the API call **http://localhost:3000/display**

When I first displayed the data without Country Name Headers
Everything displayed correctly aligned
However now that I have added the Country Name Headers
The SCHOOL NAME | LOCATION | PERSON RESPONSIBLE | CURRENT STATUS | DEPLOYMENT YEAR
are now switching from left to right.
If anyone can figure out why this is, please inform me. 
Thank you

## Related to Visualise Data #6

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [ ] I have updated any documentation

DelroyGayle committed:
Display the Table of School Data

Display the data using sample data from
client/src/pages/schools.json

berkeli commented on client/src/App.js:
```diff
@@ -4,13 +4,15 @@ import Layout from "./components/layout";
 			<Routes>
 				<Route path="/" element={<Home />} />
 				<Route path="/about/this/site" element={<About />} />
+				<Route path="/display" element={<DisplayData />} />
```
Can we rename the component do "Schools" or "DisplaySchools", we will be displaying different data all over the app so we need to give meaningful names :) Route can be called "/schools"

DelroyGayle replied:
Done

berkeli commented on client/src/pages/DisplayData.js:
```diff
@@ -0,0 +1,43 @@
+
+// Sort the Data by Countries, then by School Name, alphabetically
+
+function sortTheData(data) {
```
This is good, but we need to think about sorting the data in other ways, e.g. by number of students/teachers and etc. 

In my opinion, it should be handled in the backend

DelroyGayle replied:
Will look into doing that later as this is just a prototype to see how the display will look

berkeli commented on client/src/pages/DisplayEachCountry.js:
```diff
@@ -0,0 +1,14 @@
+import DisplayTable from "./DisplayTable.js";
+import "./DisplayData.css";
+
+const DisplayEachCountry = ({ theCountries, theData }) => {
+	const listofCountries = theCountries.map((countryName) => (
+		<div key ={countryName}>
+			<h1 className="header">{countryName}</h1>
+			<DisplayTable countryName={countryName} schoolData={theData}></DisplayTable>
+		</div>
+	));
+	return <div>{listofCountries}</div>;
+};
```
Here I think it will be best if we use one table and when it comes to country name we should merge columns `<Td colSpan={5}>` if you have 5 columns. 

Without it, the table looks uneven for each school. 

Another way to fix it would be to set a fixed width for each column, but this will be hard to make responsive (if users vistit from mobile)

DelroyGayle replied:
Used BOX instead however to little avail. Columns still not aligning properly

berkeli commented on client/src/pages/DisplayTable.js:
```diff
@@ -0,0 +1,50 @@
+			>
+			</DisplayRows>
+
+			<Tfoot>
+				<Tr>
+					<Th>&nbsp;</Th>
+					<Th>&nbsp;</Th>
+					<Th>&nbsp;</Th>
+					<Th>&nbsp;</Th>
+					<Th>&nbsp;</Th>
+				</Tr>
+			</Tfoot>
```
Do we need Tfoot with empty data? I think we can remove it

DelroyGayle replied:
Done

berkeli commented on client/src/pages/DisplayData.css:
```diff
@@ -0,0 +1,23 @@
+:root {
+  --header-colour: #37b0f0;
+  --white-colour: #ffffff;
```
We should try to use colors of the Theme in chakra so it's consistent everywhere (also configured for day/night mode).

To setup a color for day/night mode we can call a function `useColorModeValue(lightModeValue, darkModeValue)` from chakra.

We can also use colors from chakra, such as `teal.100`, and `teal.900` from light to dark.

DelroyGayle replied:
Done

berkeli reviewed (changes requested):
Thanks Delroy! I left a few minor comments

DelroyGayle committed:
Reviewers' Comments implemented

Proposed visualisation of School Data - note this is at prototype stage
The columns are not aligning properly
At this stage, still using a JSON file as the source of the data
